pkg/conditions/conditions.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package conditions
+
+import (
+	"sort"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ReadyConditionType indicates a resource is ready.
+	ReadyConditionType = "Ready"
+)
+
+// Conditions is an alias for a slice of metav1.Condition objects and provides
+// helpful functions.
+type Conditions []metav1.Condition
+
+// c4g returns a Conditions type from a Getter. This is used throughout this
+// package to reduce the need to write "Conditions(obj.GetConditions())"".
+func c4g(g Getter) Conditions {
+	return g.GetConditions()
+}
+
+// GetConditions allows the Conditions type to implement the Getter interface.
+func (l Conditions) GetConditions() []metav1.Condition {
+	return l
+}
+
+// Get returns the condition with the given type, otherwise nil is returned.
+func (l Conditions) Get(t string) *metav1.Condition {
+	for _, c := range l {
+		if c.Type == t {
+			return &c
+		}
+	}
+	return nil
+}
+
+// Has returns true if a condition with the given type exists.
+func (l Conditions) Has(t string) bool {
+	return l.Get(t) != nil
+}
+
+// IsTrue returns true if the condition with the given type exists and is
+// True, otherwise false is returned.
+func (l Conditions) IsTrue(t string) bool {
+	if c := l.Get(t); c != nil {
+		return c.Status == metav1.ConditionTrue
+	}
+	return false
+}
+
+// IsFalse returns true if the condition with the given type exists and is
+// False, otherwise false is returned.
+func (l Conditions) IsFalse(t string) bool {
+	if c := l.Get(t); c != nil {
+		return c.Status == metav1.ConditionFalse
+	}
+	return false
+}
+
+// IsUnknown returns true if the condition with the given type exists and is
+// Unknown, otherwise false is returned.
+func (l Conditions) IsUnknown(t string) bool {
+	if c := l.Get(t); c != nil {
+		return c.Status == metav1.ConditionUnknown
+	}
+	return true
```

Comment:
This should be `return false`?

Reply:
The old function returned `true`, and there's even a test that I did not touch that shows this. Perhaps it *should* return `false`, but changing it here and now could break other things. Will revisit in separate change if required.